File: src/main/java/ru/stnk/RestTestAPI/repository/VerificationCodeRepository.java
package ru.stnk.RestTestAPI.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import ru.stnk.RestTestAPI.entity.VerificationCode;

import java.util.Optional;

public interface VerificationCodeRepository extends JpaRepository<VerificationCode, Long> {
    Optional<VerificationCode> findByUserEmail (String email);
}

File: src/main/kotlin/ru/stnk/resttestapi/repository/VerificationCodeRepository.kt
package ru.stnk.resttestapi.repository

import org.springframework.data.jpa.repository.JpaRepository
import ru.stnk.resttestapi.entity.VerificationCode
import java.util.*

interface VerificationCodeRepository : JpaRepository<VerificationCode, Long> {
    fun findByUserEmail(email: String): Optional<VerificationCode>
}
